server/django_server/permissions.py

```diff
@@ -35,16 +44,29 @@ def has_permission(self, request, view):
         :return: True/False : Whether the user is a Inventory Manager or Not
         """
         user = CustomUser.objects.get(email=request.user)
+        correct_organization = [None, None, None]
+        if request.GET.get("organization", None) is not None:
+            correct_organization[0] = get_self_org_query(user, request)
+        if request.data.get('organization', None) is not None:
+            correct_organization[1] = get_self_org_body(user, request)
+        if request.parser_context['kwargs'] is not None:
+            if 'pk' in request.parser_context['kwargs']:
+                correct_organization[2] = get_self_org(user, request)
+
+        for found_false in correct_organization:
+            if found_false is not None and not found_false:
+                return False
+
+        sent_proper_organization = False
+        for found_true in correct_organization:
+            if found_true:
+                sent_proper_organization = True
 
-        if view.action in ['list']:
-            return user.role == 'IM' and str(user.organization_id) == request.GET.\
-                get("organization", '')
+        if not sent_proper_organization:
+            return False
```

Comment:
My first thought was: what's this for?
But this is actually smart. We should include a comment at the top to explain why we have this.
Maybe something along the lines of: ensuring that only the requests accounted for are being made.